### PR TITLE
forge: init at 1.0.4

### PR DESCRIPTION
**Pull request by chessai**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @chessai

### CODE REVIEW

File: pkgs/development/libraries/forge/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    mesa
+    libGLU_combined
+    glfw3
+    SDL2
+    cudatoolkit
+    arrayfire
+  ];
+
+  meta = with stdenv.lib; {
+    description = "An OpenGL interop library that can be used with ArrayFire or any other application using CUDA or OpenCL compute backend. The goal of Forge is to provide high performance OpenGL visualizations for C/C++ applications that use CUDA/OpenCL. Forge uses OpenGL >=3.3 forward compatible contexts, so please make sure you have capable hardware before trying it out";
```
Comment (c0bw3b) [inline pkgs/development/libraries/forge/default.nix]: Too long -> split it into a short `description` and a `longDescription`

File: pkgs/development/libraries/forge/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    license = licenses.bsd3;
+    homepage = https://arrayfire.com/ ;
```
Comment (c0bw3b) [inline pkgs/development/libraries/forge/default.nix]:
```suggestion
    homepage = "https://arrayfire.com/";
```

File: pkgs/development/libraries/forge/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    platforms = platforms.linux;
+    maintainers = with stdenv.lib.maintainers; [ chessai ];
```
Comment (c0bw3b) [inline pkgs/development/libraries/forge/default.nix]:
```suggestion
    maintainers = with maintainers; [ chessai ];
```

File: pkgs/development/libraries/forge/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+let
+  version = "1.0.4";
+in stdenv.mkDerivation {
+  pname = "forge";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "arrayfire";
+    repo = "forge";
+    rev = "650bf611de102a2cc0c32dba7646f8128f0300c8";
```
Comment (c0bw3b) [inline pkgs/development/libraries/forge/default.nix]: You should fetch the `v${version}` tag from upstream repo

**Review by c0bw3b — CHANGES_REQUESTED** [review]
Commit message and PR title

**Comment by chessai** [comment]
applied all suggested changes and rebased

File: pkgs/development/libraries/forge/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+let
+  version = "1.0.4";
+in stdenv.mkDerivation {
+  pname = "forge";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "arrayfire";
+    repo = "forge";
+    rev = "ref/tags/v${version}";
```
Comment (c0bw3b) [inline pkgs/development/libraries/forge/default.nix]:
```suggestion
    rev = "v${version}";
```
I'm getting `unable to checkout` when testing locally

File: pkgs/development/libraries/forge/default.nix [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    mesa
+    libGLU_combined
+    glfw3
+    SDL2
+    cudatoolkit
+    arrayfire
+  ];
+
+  meta = with stdenv.lib; {
+    description = "An OpenGL interop library that can be used with ArrayFire or any other application using CUDA or OpenCL compute backend.";
```
Comment (c0bw3b) [inline pkgs/development/libraries/forge/default.nix]: (nitpick)
Manual recommends to remove any trailing dot in the short `description`

**Comment by chessai** [comment]
remaining suggested changes added, rebased

**Review by c0bw3b — APPROVED** [review]
Result of `nix-review pr 71486` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>1 package were build:</summary>

  - forge
</details>

**Comment by chessai** [comment]
thanks @c0bw3b!

**Pull request merged**